src/tools/annotation/LengthTool.js

```diff
@@ -295,3 +302,78 @@ function _sanitizeMeasuredValue(value) {
 
   return isNumber ? parsedValue : undefined;
 }
+
+/**
+ * Check if the points lie in one or more defined regions and return the applicable
+ * pixel spacing and measurement suffix
+ *
+ * @param {*} image - image
+ * @param [] points - array of measurement points, assumed to be of length 2
+ * @returns a number or undefined
+ */
+function getMeasurementMetrics(image, points) {
+  const sequenceOfUltrasoundRegions = getUltrasoundRegions(image);
+
+  // This is based on the DICOM standard for the specification of the sequence of ultrasund regions
+  // http://dicom.nema.org/medical/dicom/current/output/chtml/part03/sect_C.8.5.5.html#table_C.8-17
+  if (
+    sequenceOfUltrasoundRegions &&
+    sequenceOfUltrasoundRegions.length > 0 &&
+    points.length === 2
+  ) {
+    // Find all regions that contain both points and then sort by total area
+    const contained = sequenceOfUltrasoundRegions
+      .filter(
+        region => region.contains(points[0]) && region.contains(points[1])
+      )
+      .sort((a, b) => a.area() - b.area());
+
+    // If no region contains both of the points, find all regions that contain either the start point or contain the end point
+    // and where both regions have the same scaling
+    if (contained.length === 0) {
+      const containsStart = sequenceOfUltrasoundRegions
+        .filter(region => region.contains(points[0]))
+        .sort((a, b) => a.area() - b.area());
+      const containsEnd = sequenceOfUltrasoundRegions
+        .filter(region => region.contains(points[1]))
+        .sort((a, b) => a.area() - b.area());
+
+      containsStart.forEach(r1 => {
+        if (contained.length > 0) {
+          return;
+        }
+
+        const match = containsEnd.find(r2 => r2.units() === r1.units());
+
+        if (match) {
+          contained.push(match);
+        }
+      });
+    }
+
+    // Assume the first discovered region is the most acceptable
+    // We do not fall back to regular pixel spacing tags as we assume that if
+    // the regions were specified, then points that exist outside of the regions
+    // have an unknown physical spacing
+    const physicalDeltaX =
+      contained.length > 0 ? contained[0].physicalDeltaX : 1;
+    const physicalDeltaY =
+      contained.length > 0 ? contained[0].physicalDeltaY : 1;
+
+    return {
+      rowPixelSpacing: physicalDeltaY,
+      colPixelSpacing: physicalDeltaY,
```

Comment:
I was implementing something similar in my app that is using 2.X version and was reviewing your PR and noticed that this should return physicalDeltaX I believe. Excited to see this PR either way once I upgrade my app to a later version.